src/com/facebook/buck/ide/intellij/IjProjectTemplateDataPreparer.java

```diff
@@ -503,6 +513,41 @@ private void addAndroidResourcePaths(
     }
   }
 
+  private void addAndroidCompilerOutputPath(
+      Map<String, Object> androidProperties,
+      IjModule module,
+      Path moduleBasePath) {
+    // This is the fallback default in case we can't find a valid build target name
+    String targetName = "src_release";
+
+    if (module.getTargets() != null) {
+      for (BuildTarget target : module.getTargets()) {
+        BuildRule buildRule = buildRuleResolver.getRule(target);
+        if ((buildRule instanceof AndroidLibrary
+            || buildRule instanceof JavaLibrary) && !target.getShortName().contains("aidl")) {
```

Comment:
[AIDL](https://developer.android.com/guide/components/aidl.html) is an Android-specific interface. since a target can have multiple source-generating rules, we want to prefer the Android/Java-Library one instead of the AIDL since it will be less important to the layout preview

if we only want to set the compiler output for android & java libs but can't deal with particular target types here, then where should that be done? (we're already setting android properties in this part of the class) here, we just want to pick the scratch folder for the module but only for android/java projects.